library/psa_crypto_mac.h

```diff
@@ -23,15 +23,93 @@
 
 #include <psa/crypto.h>
 
+/** Internal API for starting an HMAC operation, using PSA hash primitives.
```

Comment:
It seems we are quite close for this interface to be the MAC PSA driver interface (but for clone). Thus, what about to do it ? For sign setup the API would be:
```c
psa_status_t mbedtls_psa_hmac_sign_setup(
    mbedtls_psa_hmac_operation_t *operation,
    const psa_key_attributes_t *attributes,
    const uint8_t *key_buffer, size_t key_buffer_size,
    psa_algorithm_t alg )
```

Then in psa_crypto.c instead of calling this HMAC PSA driver interface, call the driver wrapper interface and you get acceleration support if any.

Reply:
I thought about that while I was doing the original work, and then decided against doing it here because it would be a pretty invasive change. Doing it for HKDF is pretty straight-forward, but the way TLS 1.2 PRF is implemented requires being able to copy the HMAC context, and the PSA MAC API did not provide for that functionality. Meaning that on driver-accelerated (H)MAC, we don't currently have a way to copy the context structure, so for TLS 1.2 PRF it would regardless have meant that we need to stick to a pure-software implementation on top of PSA hash.

And given that outcome, I believe it doesn't make sense to do it for HKDF either. Meaning that if the need is gone, it doesn't quite make sense to put in the work either.

The alternative would be to move HKDF onto the MAC API, and move TLS 1.2 PRF directly onto the hash API (without mentioning HMAC). Is that preferable?

Reply:
I did end up converting the calls to the internal HMAC interface to regular MAC calls, see the last two commits 
f9cc658 (converting HKDF) and 1878c0d (converting TLS 1.2 PRF and making the HMAC implementation static to the MAC driver).

The downside is that the context structure for TLS 1.2 PRF key derivation now contains the secret instead of the hashed secret. I don't consider this a risk, since an attacker can do as much with the hashed secret as with the actual secret. It also opens up a path for doing these operations with actual key as secret input instead of plaintext buffers, which would be good for eventually supporting e.g. an HKDF operation on an opaque key in an accelerator which only supports HMAC, or for storing the TLS 1.2 PRF secret as an opaque key.

If you don't like these, give the feedback and I can just drop them (these two commits) from the PR, in which case 562247a contains all other changes you requested (besides unification of HMAC with MAC API).